common/ManagementTestShared/Redesign/ManagementRecordedTestBase.cs

```diff
@@ -39,22 +39,19 @@ public abstract class ManagementRecordedTestBase<TEnvironment> : RecordedTestBas
         private ResourceType _resourceType;
         protected string ApiVersion { get; }
 
-        protected ManagementRecordedTestBase(bool isAsync, RecordedTestMode? mode = default, bool ignoreArmCoreDependencyVersions = true)
+        protected ManagementRecordedTestBase(bool isAsync, RecordedTestMode? mode = default)
             : base(isAsync, mode)
         {
             AdditionalInterceptors = new[] { new ManagementInterceptor(this) };
 
             SessionEnvironment = new TEnvironment();
             SessionEnvironment.Mode = Mode;
             Initialize();
-            if (ignoreArmCoreDependencyVersions)
-            {
-                IgnoreArmCoreDependencyVersions();
-            }
+            IgnoredQueryParameters.Add("api-version");
         }
 
-        protected ManagementRecordedTestBase(bool isAsync, ResourceType resourceType, string apiVersion, RecordedTestMode? mode = default, bool ignoreArmCoreDependencyVersions = true)
-            : this(isAsync, mode, ignoreArmCoreDependencyVersions)
+        protected ManagementRecordedTestBase(bool isAsync, ResourceType resourceType, string apiVersion, RecordedTestMode? mode = default)
```

Comment:
This constructor is used for tests with `api-version` input, the `api-version` matching is not ignored.